### DISCUSSION TITLE
Breaking change in `sort` attribute

### DISCUSSION THREAD

**Discussion by danieldormann** [directus/directus#5060] [Archive]
Hi everyone,

I ran into some troubles using ```sort``` when querying in my extension on **rc.54**. 

Lets assume I have a collection ```kids``` with the attribute ```name```.

Usually I'd query like this:

```
const service = new ItemsService('kids', { schema: req.schema, accountability: req.accountability });

service
  .readByQuery({ fields: ['name'], sort: 'name' })
  .then((results) => res.json(results.map((item) => { return item.name })))
  .catch((error) => {
    return next(new ServiceUnavailableException(error.message));
  });
```

this does throw a ```query.sort.map is not a function```. I had a deeper look and found this commit:

https://github.com/directus/directus/commit/2a738b7b52cafb1105cc7c51f3caaa312958e8f8#diff-2e2cf1b071c11ed335918d351c9805acf2e17c4d3ff49532a10bfc89111b9ed8

It does in fact change the expected format of ```sort```. So the working version of above query is now:

```
.readByQuery({ fields: ['name'], sort: [ { column: 'name' } ] })
```

Is this expected behavior? From my understanding it's breaking and already published in rc.54.

**Comment by rijkvanzanten** [comment]
Yeah this is expected behavior. We're still tweaking the exact final implementation details for these exposed internals (which is also why they aren't documented yet), so as long as we're in `rc`, keep an eye out for these types of changes 👍🏻

**Reply by danieldormann** [reply]
Thanks for the clarfication ✌️

**Reply by rijkvanzanten** [reply]
And if you have any thoughts on how to improve the structure of any of this in the meantime, please do let me know! This is the time to tweak and optimize, cause as soon as it's documented and released as v9.0.0 it'll be hard to change 🙂

**Reply by danieldormann** [reply]
I'll try to make a PR for a fallback check, if you agree on this.  To not have a hard break for everyone when updating -- and to prevent a bit of future community and issue management overhead 😄

**Reply by rijkvanzanten** [reply]
Hmmm I'd rather not muddy up the codebase too much with all sorts of checks and fallbacks just to smoothen out the differences between pre-release versions personally.. 🤔

**Reply by danieldormann** [reply]
I totally understand and I'll gladly follow your decisions on this, I'm really new to directus and didn't contribute anything so far. 

Just to clarify though: This isn't just breaking between pre-release versions, but also to every major release before (if I'm not mistaken 🤔 )

**Reply by danieldormann** [reply]
Oh my, I am mistaken! I didn't realize v8 is the php codebase. So: A double yes from me and sorry for the confusion :)
